Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -410,3 +469,14 @@ extension LoginDetailViewController: LoginTableViewCellDelegate {
         return false
     }
 }
+
+extension LoginDetailViewController: PasscodeEntryDelegate {
+    func passcodeValidationDidSucceed() {
+        self.removeBackgroundedBlur()
+        self.navigationController?.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    func userDidCancelValidation() {
+        self.navigationController?.popToRootViewControllerAnimated(false)
+    }
+}
```

Comment:
It looks like all of these changes to `LoginDetailViewController.swift` are exact duplicates of the changes to `LoginListViewController.swift`. Would it be worth creating a shared parent class or helper?